pkg/southbound/gnmi/conn.go

```diff
@@ -132,4 +147,66 @@ func (c *conn) ID() ConnID {
 	return c.id
 }
 
+// TargetID returns target ID associated with this connection
+func (c *conn) TargetID() topoapi.ID {
+	return c.targetID
+}
+
+// Connect connects to the target
+func (c *conn) Connect() {
+	c.clientConn.Connect()
+}
+
+// State returns connection state
+func (c *conn) State() connectivity.State {
+	return c.clientConn.GetState()
+}
+
+func (c *conn) WatchState(ctx context.Context, ch chan<- connectivity.State) error {
+	c.connStateWatchersMu.Lock()
+	c.connStateWatchers = append(c.connStateWatchers, ch)
+	c.connStateWatchersMu.Unlock()
+
+	go func() {
+		<-ctx.Done()
+		c.connStateWatchersMu.Lock()
+		connStateWatchers := make([]chan<- connectivity.State, 0, len(c.connStateWatchers)-1)
+		for _, connStateWatcher := range connStateWatchers {
+			if connStateWatcher != ch {
+				connStateWatchers = append(connStateWatchers, connStateWatcher)
+			}
+		}
+		c.connStateWatchers = connStateWatchers
+		c.connStateWatchersMu.Unlock()
+	}()
+	return nil
+
+}
+
+func (c *conn) processConnStateEvents() {
+	log.Infof("Starting processing of connection state events for connection: %s", c.id)
+	go func() {
```

Comment:
This goroutine is redundant since this function is also being called in a new goroutine already.

Reply:
Good catch.